src/net/parse.cpp

```diff
@@ -34,28 +34,67 @@
 
 namespace net
 {
+    void get_network_address_host_and_port(const std::string& address, std::string& host, std::string& port)
+    {
+        // require ipv6 address format "[addr:addr:addr:...:addr]:port"
+        if (address.find(']') != std::string::npos)
+        {
+            host = address.substr(1, address.rfind(']') - 1);
+            if ((host.size() + 2) < address.size())
+            {
+                port = address.substr(address.rfind(':') + 1);
+            }
+        }
+        else
+        {
+            host = address.substr(0, address.rfind(':'));
+            if (host.size() < address.size())
+            {
+                port = address.substr(host.size() + 1);
+            }
+        }
+    }
+
     expect<epee::net_utils::network_address>
     get_network_address(const boost::string_ref address, const std::uint16_t default_port)
     {
-        const boost::string_ref host = address.substr(0, address.rfind(':'));
+        std::string host_str = "";
+        std::string port_str = "";
+
+        bool ipv6 = false;
 
-        if (host.empty())
+        get_network_address_host_and_port(std::string(address), host_str, port_str);
+
+        boost::string_ref host_str_ref(host_str);
+        boost::string_ref port_str_ref(host_str);
+
+        if (host_str.empty())
             return make_error_code(net::error::invalid_host);
-        if (host.ends_with(".onion"))
+        if (host_str_ref.ends_with(".onion"))
             return tor_address::make(address, default_port);
-        if (host.ends_with(".i2p"))
+        if (host_str_ref.ends_with(".i2p"))
```

Comment:
This all seems to be gratuituous changes, no ?

Reply:
Seems so, but was done in order to pull get_network_address_host_and_port() (see above) into its own function, as the same logic was used elsewhere in the codebase and should not be duplicated if possible.